xarray/core/dataarray.py

```diff
@@ -1369,6 +1369,61 @@ def real(self):
     @property
     def imag(self):
         return self._replace(self.variable.imag)
+    
+    def dot( self, other):
+        """Perform sum product of two DataArrays along their shared dims. 
+        Equivalent to taking taking tensor dot over all shared dims'''
+
+        Parameters
+        ----------
+        other: the DataArray with which the DataArray calling this function 
```

Comment:
this is not quite numpydoc style. Take a look at how parameters are written: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections

This should be
```
other : DataArray
    The other array with which the dot product is performed.
```